ReactCommon/fabric/components/text/paragraph/ParagraphState.h

```diff
@@ -8,50 +8,39 @@
 #pragma once
 
 #include <react/attributedstring/AttributedString.h>
-#include <react/core/LocalData.h>
 #include <react/textlayoutmanager/TextLayoutManager.h>
 
 namespace facebook {
 namespace react {
-
-class ParagraphLocalData;
-
-using SharedParagraphLocalData = std::shared_ptr<const ParagraphLocalData>;
-
+  
 /*
- * LocalData for <Paragraph> component.
+ * State for <Paragraph> component.
  * Represents what to render and how to render.
  */
-class ParagraphLocalData : public LocalData {
+class ParagraphState final {
  public:
   /*
    * All content of <Paragraph> component represented as an `AttributedString`.
    */
   AttributedString getAttributedString() const;
-  void setAttributedString(AttributedString attributedString);
+  void setAttributedString(AttributedString const &attributedString) const;
 
   /*
    * `TextLayoutManager` provides a connection to platform-specific
    * text rendering infrastructure which is capable to render the
    * `AttributedString`.
    */
   SharedTextLayoutManager getTextLayoutManager() const;
-  void setTextLayoutManager(SharedTextLayoutManager textLayoutManager);
+  void setTextLayoutManager(SharedTextLayoutManager const &textLayoutManager) const;
 
 #ifdef ANDROID
-  folly::dynamic getDynamic() const override;
-#endif
-
-#pragma mark - DebugStringConvertible
-
-#if RN_DEBUG_STRING_CONVERTIBLE
-  std::string getDebugName() const override;
-  SharedDebugStringConvertibleList getDebugProps() const override;
+  folly::dynamic getDynamic() const;
 #endif
 
  private:
-  AttributedString attributedString_;
-  SharedTextLayoutManager textLayoutManager_;
+  mutable AttributedString attributedString_;
```

Comment:
Doing that I mentioned before (constructing fresh Data Object) will allow to avoid all that complexity.
Using `mutable` is quite dangerous, so we should try to avoid that.